app/Http/Controllers/admin/TopicController.php

```diff
@@ -0,0 +1,188 @@
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $topic = $this->topic->show($id);
+
+        return view('admin.topic.show', compact('topic'));
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $topic = $this->topic->find($id);
+
+        return view('admin.topic.edit', compact('topic'));
+    }
+
+    public function updateAll(TopicFormRequest $request, $id)
+    {
+        $topic = $this->topic->find($id);
+        $image = $request->file('image_topic');
+        $img = $this->saveImage($image);
+        $request->merge(['images' => $img]);
+        $topic->update($request->all());
+
+        return back()->with('success', trans('message.success'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $test = Topic::find($id);
+        $columnName = Input::get('name');
+        $columnValue = Input::get('value');
+        if (Input::has('name')) {
+            $test = Topic::select()
+                ->where('id', '=', $id)
+                ->update([$columnName => $columnValue]);
+
+            return response()->json(['code' => 200], 200);
+        }
+
+        return response()->json(['error' => 400, 'message' => trans('message.not_enought_params')], 400);
+    }
+
+    public function bulk_update(Request $request)
+    {
+        if (Input::has('ids_to_edit') && Input::has('bulk_name') && Input::has('bulk_value')) {
+            $ids = Input::get('ids_to_edit');
+            $bulkName = Input::get('bulk_name');
+            $bulkValue = Input::get('bulk_value');
+            foreach ($ids as $id) {
+                $test = Test::select()->where('id', '=', $id)->update([$bulkName => $bulkValue]);
+            }
+            $message = trans('message.done');
+        } else {
+            $message = trans('message.error');
+
+            return Redirect::back()->withErrors(['message' => $message])->withInput();
+        }
+
+        return Redirect::back()->with('message', $message);
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        $this->topic->delete($id);
+
+        return back()->with('sucess', __('message.sucess'));
+    }
+
+    public function saveImage($image)
+    {
+//        dd($image);
+//        save in storage
+        $filenameWithExt = $image->getClientOriginalName();
+        $filename = pathinfo($filenameWithExt, PATHINFO_FILENAME);
+        $extension = $image->getClientOriginalExtension();
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/admin/TopicController.php</br>**Line**: 183</br>Variable "new_name" is not in valid camel caps format